release/nightly_tests/setup_cluster_compute_config_updater.py

```diff
@@ -0,0 +1,147 @@
+import os
+import ray
+import asyncio
+import logging
+from typing import List, Union
+from dataclasses import dataclass
+import requests
+import anyscale
+import time
+import argparse
+from ray.util.scheduling_strategies import NodeAffinitySchedulingStrategy
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ClusterComputeConfigUpdate:
+    # Path to the field that needs to be updated
+    field_path: List[Union[str, int]]
+    # Field value to be updated to
+    field_value: Union[str, int]
+    # When to update the field in terms of seconds since run of the updater
+    update_time_seconds_since_run: int
+
+    def __init__(self, update: str):
+        field_path, field_value, update_time_seconds_since_run = update.split(":")
+
+        self.field_path = []
+        for field in field_path.split("."):
+            if field.isdigit():
+                self.field_path.append(int(field))
+            else:
+                self.field_path.append(field)
+
+        if field_value.isdigit():
+            self.field_value = int(field_value)
+        else:
+            self.field_value = field_value
+
+        self.update_time_seconds_since_run = int(update_time_seconds_since_run)
+
+
+class ClusterComputeConfig:
+    def __init__(self):
+        self.cluster_id = os.environ["ANYSCALE_CLUSTER_ID"]
+        sdk = anyscale.AnyscaleSDK()
+        self.cluster = sdk.get_cluster(self.cluster_id)
+        self.compute_config_dict = anyscale.compute_config.get(
+            name="", _id=self.cluster.result.cluster_compute_id
+        ).config.to_dict()
+        logging.info(
+            f"Fetched compute config {self.compute_config_dict} "
+            f"for cluster {self.cluster}"
+        )
+
+    def update(self, field_path: List[Union[str, int]], field_value: Union[str, int]):
```

Comment:
~~I'm confused. Why is this a list?~~ Nevermind, figured it out. Each element represents a level of nest.